packages/sc-toolkit/commands/init/assembly/contracts/main.ts

```diff
@@ -0,0 +1,26 @@
+// The entry file of your WebAssembly module.
+import {
+  callerHasWriteAccess,
+  generateEvent,
+  toBytes,
+} from '@massalabs/massa-as-sdk';
+import { Args } from '@massalabs/as-types';
+
+// This function is called when the contract is deployed.
+export function constructor(args: StaticArray<u8>): StaticArray<u8> {
+  // This line is important. It ensure that this function can't be called in the future.
```

Comment:
The best way to ensure that a function is not called in future is to not write that function :-)

Reply:
I don't know how (in a simple and comprehensive way) you can provide the constructor function and all the possibilities it give without actually writing it. We could have a naming in our SC system saying that if a function is called `constructor` then it can be called only by write access callers but Damir was against putting this kind of "conditions" in our SC system.